src/components/search/Search.js

```diff
@@ -40,48 +60,128 @@ const StyledComboboxPopover = styled(ComboboxPopover)`
 
 const Search = (props) => {
   const { setViewport } = useContext(SearchContext)
+  const isDesktop = useIsDesktop()
+  const geolocation = useGeolocation()
+  const [cityName, setCityName] = useState(null)
 
   // Hack: Reach's Combobox passes the ComboboxOption's value to handleSelect
   // So we will keep a map of the value to the place id, which handleSelect also needs
   const descriptionToPlaceId = useRef({})
 
+  const inputRef = useRef(null)
+
   const {
     ready,
     value,
     suggestions: { status, data },
     setValue,
   } = usePlacesAutocomplete()
 
+  useEffect(() => {
+    if (value === '') {
+      inputRef.current.blur()
+      inputRef.current.focus()
+    }
+  }, [value])
+
   const handleInput = (e) => {
     setValue(e.target.value)
   }
 
+  useEffect(() => {
+    async function getCityName() {
+      const city = await extractCityName({
+        lat: geolocation.latitude,
+        lng: geolocation.longitude,
+      })
+      setCityName(city)
+    }
+    getCityName()
+    // TODO: Need to debounce this so that the server doesn't get killed
+    // See: https://usehooks.com/useDebounce/
+  }, [geolocation])
+
+  let currentLocationEntry = {
+    place_id: null,
+    description: 'Current Location',
+    structured_formatting: {
+      main_text: 'Current Location',
+      secondary_text: `${cityName ? cityName : ''}`,
+    },
+  }
+
   const handleSelect = async (description) => {
     setValue(description, false)
-    const viewportBounds = await getViewportBounds(
-      descriptionToPlaceId.current[description],
-    )
+
+    let viewportBounds
+    if (description === 'Current Location') {
+      // Use fixed viewport around the lat and long of the current location
+      const { latitude, longitude } = geolocation
+
+      viewportBounds = {
+        ne: {
+          lat: latitude + CURRENT_LOCATION_VIEWPORT_RADIUS,
+          lng: longitude + CURRENT_LOCATION_VIEWPORT_RADIUS,
+        },
+        sw: {
+          lat: latitude - CURRENT_LOCATION_VIEWPORT_RADIUS,
+          lng: longitude - CURRENT_LOCATION_VIEWPORT_RADIUS,
+        },
+      }
```

Comment:
I actually think it'd help readability to make a function that takes geolocation and returns bounds around it. `getGeolocationViewportBounds`